apps/angular/crud/src/app/store/todo/todo-store.ts

```diff
@@ -0,0 +1,77 @@
+import { Injectable, inject } from '@angular/core';
+
+import {
+  injectMutation,
+  injectQuery,
+  injectQueryClient,
+} from '@tanstack/angular-query-experimental';
+import { Todo } from '../../interfaces/todo.interface';
+import { TodoService } from '../../services/todo.service';
+import { OperationType } from '../enums/actions.enum';
+import { TodoStateArgs } from './todo-args';
+@Injectable({
+  providedIn: 'root',
+})
+export class TodoStore {
+  private readonly QUERY_KEY = 'TODOS_STORE';
+  private queryClient = injectQueryClient();
+  private todoService = inject(TodoService);
+
+  public query = injectQuery(() => ({
+    queryKey: [this.QUERY_KEY],
+    queryFn: () => this.todoService.get().toPromise(),
+  }));
+
+  mutation = injectMutation(() => ({
+    mutationFn: async (args: TodoStateArgs) => {
+      switch (args.type) {
+        case OperationType.UPDATE: {
+          return await this.todoService.update(args.payload).toPromise();
+        }
+        case OperationType.DELETE: {
+          return await this.todoService.delete(args.payload).toPromise();
+        }
+        case OperationType.GET: {
+          return await this.todoService.get().toPromise();
+        }
+      }
+    },
+    onSuccess: (data, variables) => {
+      const currentTodos =
+        this.queryClient.getQueryData<Todo[]>([this.QUERY_KEY]) ?? [];
+      switch (variables.type) {
+        case OperationType.DELETE: {
+          return this.queryClient.setQueryData(
+            [this.QUERY_KEY],
+            this.deleteTodo(currentTodos, variables.payload),
+          );
+        }
+        case OperationType.UPDATE: {
+          return this.queryClient.setQueryData(
+            [this.QUERY_KEY],
+            this.updateTodo(currentTodos, variables.payload),
+          );
+        }
+        case OperationType.GET: {
+          return this.queryClient.setQueryData([this.QUERY_KEY], data);
+        }
+      }
+    },
+    onError: (error) => {
+      console.error(error);
+    },
+  }));
```

Comment:
You prefer to write it this way, than create a new function for each method ?

Reply:
Previously, I attempted a method wherein I declared an empty variable named updateTodos, subsequently modifying it within a switch case, and ultimately setting the query data. However, I found this approach unsatisfactory, thats why I did this modifications. Could you kindly recommend an alternative method for accomplishing this task?

Reply:
> You prefer to write it this way, than create a new function for each method ?

I understand your idea from your solution. I will need to create a separate mutation function for each action, such as update and delete. This approach should be very straightforward. Thank you

Reply:
I need to update my solution, but yes I think I prefer a single function per operation. Easier to maintain